src/services/completions.ts

```diff
@@ -627,6 +635,7 @@ namespace ts.Completions {
         formatContext: formatting.FormatContext,
         preferences: UserPreferences,
         cancellationToken: CancellationToken,
+        autoImportProvider: Program | undefined,

```

Comment:
I think instead of passing autoImportProvider, completions should get it when using to avoid having to update/get it in cases when the completion is for eg class method or something that doesnt need auto imports

Reply:
👍 this is possible now that it’s exposed on `LanguageServiceHost` instead of created in the services layer.